pkg/asset/cluster/azure/azure.go

```diff
@@ -50,8 +53,21 @@ func tagVNet(ctx context.Context, clusterID string, installConfig *installconfig
 	}
 
 	resourceGroupName := installConfig.Config.Azure.NetworkResourceGroupName
+	clientOpts := &arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: cloud.Configuration{
+				ActiveDirectoryAuthorityHost: session.Environment.ActiveDirectoryEndpoint,
+				Services: map[cloud.ServiceName]cloud.ServiceConfiguration{
+					cloud.ResourceManager: {
+						Audience: session.Environment.TokenAudience,
+						Endpoint: session.Environment.ResourceManagerEndpoint,
+					},
+				},
+			},
+		},
+	}
```

Comment:
The session has the cloud config saved:
```suggestion
	clientOpts := &arm.ClientOptions{
		ClientOptions: azcore.ClientOptions{
			Cloud: session.CloudConfig,
		},
	}
```

Reply:
done, thanks